### PR TITLE
added PasswordSSO docs

### PR DESCRIPTION
**Pull request by kalilasimpson**

### CODE REVIEW

**Comment by Lauragra** [comment]
@kalilasimpson , please follow the Get started steps in this topic: https://msgo.azurewebsites.net/add/document/manage-content/get-started-with-github.html. I've added the Do not merge label to your PR. Please let me know when your content is ready for an editorial review. When you're ready to publish, set the Ready to merge label.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+Create password single sign on credentials for a user or group by id.
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | Users can create credentials for themselves. Service Principal Owners and admins with the following roles will be able to create credentials for any user or group: GlobalAdministrator, ApplicationAdministrator, CloudApplicationAdministrator. Anyone with microsoft.directory/servicePrincipals/managePasswordSingleSignOnCredentials RBAC role will be able to create credentials for any user or group.|
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: This needs to be an application permission in this table.
You can add a note under the table, that for the delegated permission, what roles user must be in.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+| Authorization | Bearer {token} |
+
+## Request body
+
+In the request body, provide a JSON object with the following parameters.
+
+| Parameter    | Type        | Description |
+|:-------------|:------------|:------------|
+|id|String|The id of the user or group this credential set belongs to.|
+|credentials|credential collection|A list of credential objects that define the complete login flow.|
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: credential needs to link to the credential resource definition.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+| Delegated (work or school account)     | Users can create credentials for themselves. Service Principal Owners and admins with the following roles will be able to create credentials for any user or group: GlobalAdministrator, ApplicationAdministrator, CloudApplicationAdministrator. Anyone with microsoft.directory/servicePrincipals/managePasswordSingleSignOnCredentials RBAC role will be able to create credentials for any user or group.|
+| Delegated (personal Microsoft account) | Not supported. |
+| Application                            | Application.ReadWrite.All, Directory.AccessAsUser.All, Directory.ReadWrite.All |
+
+## HTTP request
+
+<!-- { "blockType": "ignored" } -->
+
+```http
+POST /servicePrincipals/createPasswordSingleSignOnCredentials
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: Shouldn't this be:  
`POST /servicePrincipals/{id}/createPasswordSingleSignOnCredentials`

And same below?

This is creating an SSO credential set for this user/group, for this app.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+author: "bharathramh92"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# servicePrincipal: createPasswordSingleSignOnCredentials
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Create password single sign on credentials for a user or group by id.
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: Do you need to say that these credentials are to allow a user or group (members) to SSO to this servicePrincipal

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+In the request body, provide a JSON object with the following parameters.
+
+| Parameter    | Type        | Description |
+|:-------------|:------------|:------------|
+|id|String|The id of the user or group this credential set belongs to.|
+|credentials|credential collection|A list of credential objects that define the complete login flow.|
+
+## Response
+
+If successful, this method returns `200, OK` response code and a new [passwordSingleSignOnCredentialSet](../resources/passwordsinglesignoncredentialset.md) object in the response body.
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: Remove the comma from 200, OK.  This is a doc generation bug.

File: api-reference/beta/api/serviceprincipal-deletepasswordsinglesignoncredentials.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+In the request body, provide a JSON object with the following parameters.
+
+| Parameter    | Type        | Description |
+|:-------------|:------------|:------------|
+|id|String|The id of the user or group this credential set belongs to.|
+
+## Response
+
+If successful, this method returns `200, OK` response code. It does not return anything in the response body.
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-deletepasswordsinglesignoncredentials.md]: 200, OK -> 200 OK

File: api-reference/beta/api/serviceprincipal-getpasswordsinglesignoncredentials.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+In the request body, provide a JSON object with the following parameters.
+
+| Parameter    | Type        | Description |
+|:-------------|:------------|:------------|
+|id|String|The id of the user or group this credential set belongs to.|
+
+## Response
+
+If successful, this method returns `200, OK` response code and a new [passwordSingleSignOnCredentialSet](../resources/passwordsinglesignoncredentialset.md) object in the response body.
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-getpasswordsinglesignoncredentials.md]: 200, OK -> 200 OK

File: api-reference/beta/api/serviceprincipal-updatepasswordsinglesignoncredentials.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+In the request body, provide a JSON object with the following parameters.
+
+| Parameter    | Type        | Description |
+|:-------------|:------------|:------------|
+|id|String|The id of the user or group this credential set belongs to.|
+|credentials|credential collection|A list of credential objects that define the complete login flow.|
+
+## Response
+
+If successful, this method returns `200, OK` response code. It does not return anything in the response body.
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-updatepasswordsinglesignoncredentials.md]: 200, OK -> 200 OK

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | Application.ReadWrite.All, Directory.AccessAsUser.All, Directory.ReadWrite.All |
+| Delegated (personal Microsoft account) | Not supported. |
+| Application                            | Application.ReadWrite.All, Directory.AccessAsUser.All, Directory.ReadWrite.All |
+
+Please note, users can create credentials for themselves. Service Principal Owners and admins with the following roles will be able to create credentials for any user or group: GlobalAdministrator, ApplicationAdministrator, CloudApplicationAdministrator.
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: Nit: You should link to the Directory Roles topic: https://docs.microsoft.com/azure/active-directory/users-groups-roles/directory-assign-admin-roles#available-roles

File: api-reference/beta/api/serviceprincipal-deletepasswordsinglesignoncredentials.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+### Request
+
+The following is an example of the request.
+<!-- {
+  "blockType": "request",
+  "name": "serviceprincipal_deletepasswordsinglesignoncredentials"
+}-->
+
+```http
+POST https://graph.microsoft.com/beta/servicePrincipals/deletePasswordSingleSignOnCredentials
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-deletepasswordsinglesignoncredentials.md]: id is missing again

File: api-reference/beta/api/serviceprincipal-getpasswordsinglesignoncredentials.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+### Request
+
+The following is an example of the request.
+<!-- {
+  "blockType": "request",
+  "name": "serviceprincipal_getpasswordsinglesignoncredentials"
+}-->
+```http
+POST https://graph.microsoft.com/beta/servicePrincipals/getPasswordSingleSignOnCredentials
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-getpasswordsinglesignoncredentials.md]: id is missing again

File: api-reference/beta/resources/credential.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Indicates a single credential used for login to an application. For example, username is one credential, password is another credential.
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|fieldId|String|The name of the field for this credential. e.g, username or password or phoneNumber. This is defined by the application. Must match what is in the html field on singleSignOnSettings/password object.|
+|type|String|The value for this credential. e.g, mysuperhiddenpassword.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/credential.md]: You have the descriptions backwards :)
You should indicate that the value for a password type credential is write-only, and can never be read back.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
+---
+title: "servicePrincipal: createPasswordSingleSignOnCredentials"
+description: "Create password single sign on credentials for a user or group by id."
```
Comment (davidmu1) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: Might be better as (note the hyphen in sign-on): Create single sign-on credentials using a password for a user or group.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+author: "bharathramh92"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# servicePrincipal: createPasswordSingleSignOnCredentials
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Create password single sign on credentials.
```
Comment (davidmu1) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: Seems like there might be more of a description here.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | Application.ReadWrite.All, Directory.AccessAsUser.All, Directory.ReadWrite.All |
```
Comment (bharathramh92) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: Need `Directory.Read.All` if using `Application.ReadWrite.All`.

File: api-reference/beta/api/serviceprincipal-deletepasswordsinglesignoncredentials.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+Delete password single sign on credentials.
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | Application.ReadWrite.All, Directory.AccessAsUser.All, Directory.ReadWrite.All |
```
Comment (bharathramh92) [inline api-reference/beta/api/serviceprincipal-deletepasswordsinglesignoncredentials.md]: Need `Directory.Read.All` if using `Application.ReadWrite.All`.

File: api-reference/beta/api/serviceprincipal-updatepasswordsinglesignoncredentials.md [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+
+Update password single sign on credentials.
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | Application.ReadWrite.All, Directory.AccessAsUser.All, Directory.ReadWrite.All |
```
Comment (bharathramh92) [inline api-reference/beta/api/serviceprincipal-updatepasswordsinglesignoncredentials.md]: Need `Directory.Read.All` if using `Application.ReadWrite.All`.

File: api-reference/beta/api/serviceprincipal-getpasswordsinglesignoncredentials.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+
+Get a list of password single sign on credentials.
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | Application.ReadWrite.All, Directory.AccessAsUser.All, Directory.ReadWrite.All |
```
Comment (bharathramh92) [inline api-reference/beta/api/serviceprincipal-getpasswordsinglesignoncredentials.md]: Need `Directory.Read.All` if using `Application.ReadWrite.All`.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md (hunk 15 above)
Comment (dkershaw10) [reply api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: @bharathramh92 That just seems wrong.  Is that an implementation requirement?  If the dev is just changing a property on servicePrincipal, then Application.ReadWrite.All should be sufficient on its own.

Comment (bharathramh92) [reply api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: We are using azure ad graph api for this MS Graph api. They don't support `Application.ReadWrite.All` yet for delegate permission. `Directory.Read.All` requirement should go away sooner once they onboard to it.

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 19]
```diff
@@ -0,0 +1,129 @@
...
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | Application.ReadWrite.All (also needs Directory.Read.All), Directory.AccessAsUser.All, Directory.ReadWrite.All |
+| Delegated (personal Microsoft account) | Not supported. |
+| Application                            | Application.ReadWrite.All (also needs Directory.Read.All), Directory.AccessAsUser.All, Directory.ReadWrite.All |
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: Remove Directory.AccessAsUser.All - it's only a delegated permission

File: api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md [hunk 20]
```diff
@@ -0,0 +1,129 @@
...
+
+```http
+POST /servicePrincipals/{id}/createPasswordSingleSignOnCredentials
+```
+
+## Request headers
+
+| Name          | Description   |
+|:--------------|:--------------|
+| Authorization | Bearer {token} |
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-createpasswordsinglesignoncredentials.md]: For POST and PATCH/PUT requests you need an extra row in the table:

| Content-Type | application/json |

File: api-reference/beta/resources/credential.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+author: "bharathramh92"
+ms.prod: "microsoft-identity-platform"
+doc_type: "resourcePageType"
+---
+
+# credential resource type
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Indicates a single credential used for login to an application. For example, username is one credential, password is another credential.
```
Comment (dkershaw10) [inline api-reference/beta/resources/credential.md]: login or sign-in?

File: api-reference/beta/api/serviceprincipal-deletepasswordsinglesignoncredentials.md [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+
+The following is an example of the response.
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.None"
+} -->
+
+```http
+HTTP/1.1 204
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-deletepasswordsinglesignoncredentials.md]: HTTP/1.1 204 No Content

API Doctor is complaining about this too.

File: api-reference/beta/api/serviceprincipal-updatepasswordsinglesignoncredentials.md [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+
+The following is an example of the response.
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.None"
+} -->
+
+```http
+HTTP/1.1 204
```
Comment (dkershaw10) [inline api-reference/beta/api/serviceprincipal-updatepasswordsinglesignoncredentials.md]: HTTP/1.1 204 No Content

API Doctor is complaining about this too.

**Comment by dkershaw10** [comment]
@Lauragra @CelesteDG Please take a look ASAP.  This service is already live, so this doc needs publishing (if it's ready)

**Pull request merged**